### PR TITLE
feat(frontend): Add version description to pipelineVersionList. Part of #6256

### PR DESCRIPTION
**Pull request by casassg**
**Description of your changes:**
Add version description to Pipeline Version list.

had to update snapshot for test. Not sure if I did that correctly

Also added some mocked data to more easily check UI looked good.

### CODE REVIEW

File: frontend/src/pages/PipelineVersionList.tsx [hunk 1]
```diff
@@ -85,17 +91,18 @@ class PipelineVersionList extends React.PureComponent<
...
       },
+      { label: 'Description', flex: 3, customRenderer: descriptionCustomRenderer },
       { label: 'Uploaded on', flex: 1, sortKey: PipelineVersionSortKeys.CREATED_AT },
     ];
 
     const rows: Row[] = this.state.pipelineVersions.map(r => {
       const row = {
         id: r.id!,
-        otherFields: [r.name, formatDateString(r.created_at)] as any,
+        otherFields: [r.name, r.description!, formatDateString(r.created_at)] as any,
```
Comment (zijianjoy) [inline frontend/src/pages/PipelineVersionList.tsx:105]: description is optional, so I think this non-null assertion operator `!` will cause runtime failure when the description doesn't exist.

File: frontend/src/pages/PipelineVersionList.tsx [hunk 2]
```diff
@@ -38,6 +39,11 @@ interface PipelineVersionListState {
   pipelineVersions: ApiPipelineVersion[];
 }
 
+const descriptionCustomRenderer: React.FC<CustomRendererProps<string>> = (
+  props: CustomRendererProps<string>,
+) => {
+  return <Description description={props.value || ''} forceInline={true} />;
```
Comment (zijianjoy) [inline frontend/src/pages/PipelineVersionList.tsx:45]: Adding Tooltip to description and version name allows user to see the full string when hovering over to them:

https://github.com/kubeflow/pipelines/blob/master/frontend/src/pages/PipelineList.tsx#L196

Comment (casassg) [reply frontend/src/pages/PipelineVersionList.tsx]: Added. Had to add an extra span due to ToolTip binding not working for Description element. Adding span solved it.

**Comment by zijianjoy** [comment]
/lgtm
/approve

Thank you so much @casassg for adding pipeline version description!

**Pull request merged**